pkgs/development/haskell-modules/make-package-set.nix

```diff
@@ -235,12 +231,14 @@ in package-set { inherit pkgs stdenv callPackage; } self // {
       , overrides ? self: super: {}
       , modifier ? drv: drv
       , returnShellEnv ? pkgs.lib.inNixShell }:
-      let drv =
-        (extensible-self.extend
-           (pkgs.lib.composeExtensions
-              (self.packageSourceOverrides source-overrides)
-              overrides))
-        .callCabal2nix name root {};
+      let
+        extendFix' = f: fixed: fix' (extends f fixed.__unfix__);
```

Comment:
Would you be able to add a comment here on what `extendFix'` is doing?  It is not immediately obvious to me what it is doing, or why it would be needed?

Reply:
> We would like to call `extend` on `self`, but that is not available. However for fixed points that were taken using `fix'` we can get the same effect by calling `extends` on the `__unfix__` attribute and taking the fixed point again.

Would that be OK?